src/persist-client/src/cache.rs

```diff
@@ -368,10 +534,18 @@ impl StateCache {
                     let state = init_once
                         .get_or_try_init::<Box<CodecMismatch>, _, _>(|| async {
                             let init_res = init_fn().await;
+                            let token = PubSubSubscriptionToken {
+                                shard_id: shard_id.clone(),
+                                metrics: Arc::clone(&self.metrics),
+                                pubsub_sender: self.pubsub_sender.clone(),
+                            };
                             let state = Arc::new(LockingTypedState {
```

Comment:
nit: feels like this has gotten complicated enough to pull out into a `LockingTypedState::new` method